src/libraries/Microsoft.PowerFx.Core/Public/Values/CompileTimeTypeWrapperRecordValue.cs

```diff
@@ -24,5 +27,44 @@ private CompileTimeTypeWrapperRecordValue(RecordType type, RecordValue inner)
             : base(IRContext.NotInSource(type), inner.Fields)
         {
         }
+
+        protected override bool TryGetField(FormulaType fieldType, string fieldName, out FormulaValue result)
+        {
+            if (Type.TryGetFieldType(fieldName, out _))
+            {
+                // Only return field which were specified via the expectedType (IE RecordType),
+                // because inner record value may have more fields than the expected type.
+                return _fields.TryGetValue(fieldName, out result);
+            }
+
+            result = default;
+            return false;
+        }
+
+        public override async Task<DValue<RecordValue>> UpdateFieldsAsync(RecordValue changeRecord, CancellationToken cancellationToken)
```

Comment:
>UpdateFieldsAsync [](http://example.com/codeflow?start=56&length=17)

Can this be merged with base class more?

Reply:
done.